Migrate Contributing Docs to the Docs Site

KVSRoyal opened the pull request:
<!-- 

Describe what's changing, why, and any other background info.

Make sure to add:
  - Tests
  - Documentation Updates

-->

> [!NOTE]
> ✨ AI-aided PR. I used Cursor for a lot of the tedious copy-pasta-ing. Reviewers should keep an eye out for common AI foibles.

Our contributing documentation currently lives in a lot of different places. Use Cursor to migrate these to the docs site in a new "Contributing" section. The point of this is to improve the onboarding experience by centralizing our docs in one findable location.

### Before
 Docs scattered in various places:
```
📁 player/
├── 📄 CONTRIBUTING.md (comprehensive guide - ~100+ lines)
├── 📄 newCONTRIBUTORS.md (first-time contributor guide)
├── 📄 DEV_GUIDE.md (TypeScript/JavaScript development)
├── 📁 android/demo/
│   └── 📄 README.md (Android development guide)
├── 📁 ios/
│   └── 📄 README.md (iOS development guide)
└── 📁 docs/site/src/content/docs/
    └── (no contributing documentation)

📖 Wiki > iOS Development Guide
```

### After
Everything consolidated in the `contributing` section.
```
📁 player/
├── 📄 CONTRIBUTING.md (simple stub with links - 18 lines)
└── 📁 docs/site/src/content/docs/contributing/
    ├── 📄 index.mdx (main contributing guide - 109 lines)
    ├── 📄 new-contributors.mdx (first-time contributors - 193 lines)
    ├── 📄 android.mdx (Android development - 67 lines)
    ├── 📄 ios.mdx (iOS development - 354 lines)
    ├── 📄 core-web.mdx (Core & Web development - 205 lines)
    └── 📄 troubleshooting.mdx (real issues only - 77 lines)
```

Also:
1. Document a new caching issue affecting iOS builds and how to fix it. Add this to Troubleshooting.
2. Introduce a Troubleshooting section for common issues.
3. Leverage Astro components in our migrated docs. I asked Cursor to use components wherever they seemed appropriate.

### Impact
- This will result in conflicts with https://github.com/player-ui/player/pull/700, which also updates our documentation.
- We should delete the Wiki once this PR is merged.

### Future Work

These are items we should discuss and align on in a future PR. These came up during PR review.

1. [Thread](https://github.com/player-ui/player/pull/705#discussion_r2281011851). Remove GitHub CLI suggestions.
2. [Thread 1](https://github.com/player-ui/player/pull/705#discussion_r2282535621), [Thread 2](https://github.com/player-ui/player/pull/705#discussion_r2282540694). Document forks vs branches.
3. [Thread](https://github.com/player-ui/player/pull/705#discussion_r2283047074). Upgrade Android versions to newer.
4. [Thread](https://github.com/player-ui/player/pull/705#discussion_r2283049052). Users are probably okay to have multiple SDK versions installed.

### Change Type (required)
Indicate the type of change your pull request is:

<!-- 
  We use semantic versioning: https://semver.org/. Review that documentation for 
  more detailed guidelines.
-->
- [ ] `patch`
- [ ] `minor`
- [ ] `major`
- [x] `N/A`


### Does your PR have any documentation updates?
- [x] Updated docs
- [ ] No Update needed
- [ ] Unable to update docs
<!--
In an effort to standardize our process and code, please make sure you include documentation and/or update any existing documentation.
Please refer to our site https://player-ui.github.io/latest/about, and include any neccesary information that would be helpful to coders, developers, and learners.

If you are unable to update the current documents, please create an issue for us to get back to it.

-->

<!--
  To include release notes in the automatic changelong, just add a level 1 markdown header below
  and include any markdown notes to go into the changelog: https://intuit.github.io/auto/docs/generated/changelog#additional-release-notes

  Example:

  # Release Notes
  Added new plugin, to use it:
  ```typescript
  const plugin = new Plugin(...)
  ```
-->

KetanReddy commented on docs/site/src/content/docs/contributing/new-contributors.mdx:
```diff
@@ -0,0 +1,192 @@
+
+<FileTree>
+- player/
+  - android/          # Android-specific code
```
just to line it up

KetanReddy commented on CONTRIBUTING.md:
```diff
@@ -1,105 +1,18 @@
-
-## Requirements
-
-- [bazelisk](https://github.com/bazelbuild/bazelisk)
-- [pnpm >= 9.0.0](https://pnpm.io/installation)
-
-- [Swift >= 5.5](https://www.swift.org/download/)
-- [Xcode 15.3](https://developer.apple.com/download/all/)
-
-- [Android NDK 27.1.12297006](https://github.com/android/ndk/releases/tag/r27b). You'll need to add `ANDROID_NDK_HOME` to your environment manually. It will look like `export ANDROID_NDK_HOME=/Users/{USERNAME}/Library/Android/sdk/ndk/27.1.12297006`
-- Python < 3 (recommended 2.7.18) - you can use [pyenv](https://realpython.com/intro-to-pyenv/) to manage different python versions with ease.
-
-- [Signed Commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification). For convenience it is recommended to set git to sign all commits by default as mentioned [here](https://docs.github.com/en/authentication/managing-commit-signature-verification/telling-git-about-your-signing-key)
```
IMO these instructions should be preserved somewhere

KetanReddy commented on docs/site/src/content/docs/contributing/new-contributors.mdx:
````diff
@@ -0,0 +1,192 @@
+   When you're ready to submit a Pull Request, there's a great CLI command for that too:
+
+   ```bash
+   gh pr create
````
Unless we're going to recommend people download the github cli we probably don't want to have those specific commands here

kharrop reviewed (commented):
Wow! Thanks for a great contributing docs overhaul. Added some minor notes. Question, are you using markdownlint? I can't remember if that's part of this repo.

KVSRoyal committed:
Migrate contributing docs to docs site

+ Use more astro components, remove AI content
+ Document how to fix cache corruption

KVSRoyal committed:
Fix AI issue; restore deps example

KVSRoyal commented:
Lots of content changes requested. Moving this PR back to draft because I don't have time to add them.
